version: '3'
services:
  foodtrack-db:
    image: ${IMAGE}
    ports:
      - 5300:5301
    restart: on-failure 
    networks:
      - web
    labels:
      - traefik.enable=true
      - traefik.backend=foodtrack-db 
      - traefik.docker.network=web

      - traefik.http.services.foodtrack-db-grpc.loadbalancer.server.scheme=h2c 
      - traefik.http.services.foodtrack-db-grpc.loadbalancer.server.port=5301

      - traefik.http.routers.foodtrack-db-grpc.rule=Host(`db.qcenar.com`)
      - traefik.http.routers.foodtrack-db-grpc.entrypoints=grpc
      - traefik.http.routers.foodtrack-db-grpc.tls=true
      - traefik.http.routers.foodtrack-db-grpc.tls.certresolver=le 
  foodtrack-postgres:
    image: postgres 
    restart: on-failure
    environment:
      - POSTGRES_USER=${POSTGRES_USER} 
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - PGDATA=/var/lib/postgresql/data/pgdata
      - POSTGRES_DB=foodtrack
    volumes:
      - ${DATA}/postgres_data:/var/lib/postgresql
    ports:
      - 5432:5432
networks:
  web:
    external: true
  internal:
    external: false